Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -40,6 +40,24 @@ module Livecheck
       T.must(@livecheck_strategy_names).freeze
     end
 
+    sig { returns(T::Hash[T::Class[T.anything], T::Array[Symbol]]) }
+    private_class_method def self.livecheck_find_versions_parameters
+      return T.must(@livecheck_find_versions_parameters) if defined?(@livecheck_find_versions_parameters)
+
+      # Cache strategy `find_versions` method parameters, to avoid repeating
+      # this work
+      @livecheck_find_versions_parameters = T.let({}, T.nilable(T::Hash[T::Class[T.anything], T::Array[Symbol]]))
+      Strategy.constants.sort.each do |const_symbol|
```

Comment:
Does it make sense to calculate all of these at once, since we're doing individual lookups below? It seems like it might be cleaner to pass the class as a parameter to the method instead (moot if we define the lookup on strategies as described above).

Reply:
The code in `livecheck_strategy_names` was originally inlined in the livecheck command and it was eventually spun out into a method without modifying how it works. Collecting information on all strategies at once made sense in the original context but I agree that it doesn't make sense based on how it's used now. [I copied the logic in `livecheck_find_versions_parameters`, so it inherited the same unusual setup.]

Let me see if I can rework these methods to only do work as it's needed (and cache the result).